rero_ils/modules/utils.py

```diff
@@ -1095,6 +1095,36 @@ def remove_user_name(sender, user):
         del session['user_name']
 
 
+def set_session_patrons(sender, user):
+    """Set some data from patrons in the current session."""
+    from .patrons.api import current_patrons
+    if len(current_patrons) > 0:
+        s_patrons = {
+            'show_info': False,
+            'data': {}
+        }
+        for patron in current_patrons:
+            if (patron.is_blocked or patron.is_expired) \
+                    and not s_patrons['show_info']:
```

Comment:
you should remove the `not s_patrons['show_info']` test : if it's already true it's not a problem (test is more time consuming than assignation)